### PR TITLE
EDD-6: Sends an EDL token on download requests

### PR DESCRIPTION
**Pull request by macrouch**
# Overview

### What is the feature?

Adds support for EDL tokens to be included with download requests.
Refactors `startNextDownload` logic to be more reliable

### What is the Solution?

Uses `webContents.session.webRequest.onBeforeSendHeaders` to attach an `Authorization` header to the downloads.

### What areas of the application does this impact?

Downloads of files behind EDL authentication

# Testing

Testable only with ASF collection currently due to EDL groups
http://localhost:8080/projects?p=C1214470488-ASF!C1214470488-ASF&pg[1][v]=t&pg[1][gsk]=-start_date&pg[1][m]=download&qt=2023-03-26T21%3A08%3A28.467Z%2C2023-03-26T21%3A51%3A06.768Z&tl=1679857448.269!1!!

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have bumped the `version` field in package.json and ran `npm install`

### CODE REVIEW

File: src/main/eventHandlers/willDownloadEvents/onUpdated.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // @ts-nocheck
 
 import downloadStates from '../../../app/constants/downloadStates'
+// import startNextDownload from './startNextDownload'
```
Comment (macrouch) [inline src/main/eventHandlers/willDownloadEvents/onUpdated.ts:4]: Remove

File: src/main/main.ts [hunk 2]
```diff
@@ -58,6 +63,11 @@ const createWindow = async () => {
       preload: path.join(__dirname, 'preload.js')
     }
   })
+  authWindow = new BrowserWindow({
```
Comment (macrouch) [inline src/main/main.ts:66]: Not using anymore

File: src/main/utils/database/EddDatabase.ts [hunk 3]
```diff
@@ -125,6 +122,40 @@ class EddDatabase {
     return this.db('files').select().where(where).first()
   }
 
+  /**
+   * Returns the file that is in the waitingForAuth state, and matches the domain provided
+   * @param {String} domain URL to use in the whereLike clause
+   */
+  async getFileWaitingForAuthWhereUrlLike(domain) {
```
Comment (macrouch) [inline src/main/utils/database/EddDatabase.ts:129]: Remove

**Review by jaybarra — APPROVED** [review]

File: src/main/eventHandlers/__tests__/openUrl.test.ts [hunk 4]
```diff
@@ -1,50 +1,106 @@
+// @ts-nocheck
```
Comment (trevorlang) [inline src/main/eventHandlers/__tests__/openUrl.test.ts:1]: We want this?

**Review by trevorlang — COMMENTED** [review]
Code is all looking good. I will get it running locally Monday morning and give you a ✅

**Review by trevorlang — APPROVED** [review]
This looks really solid. I think all we need is the "cancel" action on the download while the item is in the authenticating state.

**Pull request merged**